__tests__/unit/shifter.ts

```diff
@@ -0,0 +1,215 @@
+        expect(mapping[country]).toEqual(expect.arrayContaining([agent.axios.defaults.httpAgent.proxy.port]));
+      });
+
+      it('create a proxy agent having a port matching several of the options in the mapping', () => {
+        const countries: EShifterCountry[] = [EShifterCountry.FRANCE, EShifterCountry.BELGIUM];
+        const mappedPorts: number[] = countries.map((c) => mapping[c]).reduce((acc, c) => [...acc, ...c], []);
+        const agent: Shifter = shifter.setIp({ countries: countries });
+
+        expect(mappedPorts).toEqual(expect.arrayContaining([agent.axios.defaults.httpsAgent.proxy.port]));
+        expect(mappedPorts).toEqual(expect.arrayContaining([agent.axios.defaults.httpAgent.proxy.port]));
+      });
+    });
+  });
+
+  describe('#strategy', () => {
+    describe('when strategy is set to CHANGE_IP_EVERY_REQUESTS', () => {
+      let shifter: Shifter;
+
+      beforeEach(() => {
+        shifter = new Shifter({ proxy, strategy });
+      });
+
+      const response1 = {
+        ip: '184.174.62.231',
+        country: 'FR',
+        asn: {
+          asnum: 9009,
+          org_name: 'M247 Ltd',
+        },
+        geo: {
+          city: 'Paris',
+          region: 'IDF',
+          region_name: 'Île-de-France',
+          postal_code: '75014',
+          latitude: 48.8579,
+          longitude: 2.3491,
+          tz: 'Europe/Paris',
+          lum_city: 'paris',
+          lum_region: 'idf',
+        },
+      };
+
+      const response2 = {
+        ip: '178.171.89.101',
+        country: 'ES',
+        asn: {
+          asnum: 9009,
+          org_name: 'M247 Ltd',
+        },
+        geo: {
+          city: 'Madrid',
+          region: 'MD',
+          region_name: 'Madrid',
+          postal_code: '28001',
+          latitude: 40.4167,
+          longitude: -3.6838,
+          tz: 'Europe/Madrid',
+          lum_city: 'madrid',
+          lum_region: 'md',
+        },
+      };
+
+      describe('using HTTP', () => {
+        it("changes the caller's IP address every request", async () => {
+          const request1Mock = nock('http://lumtest.com').get('/myip.json').once().reply(200, response1);
+          const request2Mock = nock('http://lumtest.com').get('/myip.json').once().reply(200, response2);
+
+          const result1 = await shifter.fetch({
+            method: 'get',
+            baseURL: 'http://lumtest.com',
+            url: '/myip.json',
+          });
+
+          const result2 = await shifter.fetch({
+            method: 'get',
+            baseURL: 'http://lumtest.com',
+            url: '/myip.json',
+          });
+
+          expect(result1.data).toStrictEqual(response1);
+          expect(result2.data).toStrictEqual(response2);
+
+          request1Mock.done();
+          request2Mock.done();
+        });
+      });
+
+      describe('using HTTPS', () => {
+        it("changes the caller's IP address every request", async () => {
```

Comment:
i guess its how its done in the other proxy providers butt i dont understand how is that testing tthat the IP address changes on each call

Reply:
I wrote that test because it is present in the other ones. Since everything is mocked, it is not of great use, I agree. However, what it tests is that the change of proxy IP is passed throughout to axios on 2 successive calls.